mcs/class/Mono.Compiler/Mono.Compiler.BigStep/BigStep.cs

```diff
@@ -280,6 +353,7 @@ ArgStackValue Push (Env env, Builder builder, BSType t)
 			var v = builder.EmitAlloca (t, "stack-slot");
 			var a = new ArgStackValue ();
 			a.Ptr = v;
+			a.LoweredType = t;
```

Comment:
not sure if this is okay?

Reply:
Yea I was going to add it, but didn't need it yet.

Bikeshed: "Lowered Type" is the low-level (ie LLVM) type.  `BSType` is... both the IL Type and the LLVM Type tied together.  So maybe... `a.StoredType` or something... since it's the type of the storage of the argument, not the type the pointer in `a.Ptr`?

Reply:
much better, updated PR 👍